### PR TITLE
Lift case-sensitive restriction on search / filter.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes: https://github.com/flutter/devtools/issues/805

### CODE REVIEW

File: packages/devtools/lib/src/popup.dart [hunk 1]
```diff
@@ -616,7 +622,14 @@ class AutoCompleteMatcher<T> {
     lastMatchingItems ??= matchingState[''];
 
     final List<T> matchingItems = lastMatchingItems
-        .where((T item) => item.toString().lastIndexOf('$charsToMatch') >= 0)
+        .where((T item) =>
+            item
+                .toString()
+                // Convert strings to lowercase so that matching is not case
```
Comment (natebosch) [inline packages/devtools/lib/src/popup.dart]: [optional] Instead of repeating the code _and_ giving the reason, could shorten this to the reason alone.

```
// Case insensitive matching
.toLowerCase()
.lastIndexOf(...)
```

Similar elsewhere.

Comment (kenzieschmoll) [reply packages/devtools/lib/src/popup.dart]: done.
